cyences_app_for_splunk/default/savedsearches.conf

```diff
@@ -4984,5 +5061,82 @@ action.cyences_send_email_action = 1
+action.cyences_send_email_action = 1
+action.cyences_notable_event_action.products = Google Workspace
+
+
+[Google Workspace - Suspicious File Shared by External User]
+disabled = 1
+enableSched = 1
+alert.track = 1
+alert.severity = 4
+alert.suppress = 0
+counttype = number of events
+quantity = 0
+relation = greater than
+cron_schedule = 57 * * * *
+description = The alert will generate if Suspicious File Shared by External User. \
+NOTE: cs_email_user_domain macro must be configured with the domain name in order to use this alert. \
+Data Collection - Google drive activity reports data from Splunk Add-on for Google Workspace (https://splunkbase.splunk.com/app/5556) 
+dispatch.earliest_time = -6h@h
+dispatch.latest_time = now
+display.general.type = statistics
+display.page.search.tab = statistics
+display.page.search.mode = fast
+request.ui_dispatch_app = cyences_app_for_splunk
+request.ui_dispatch_view = search
+search = `cs_gsuite` sourcetype="gws:reports:drive" NOT email IN ("", "null") `cs_email_suspicious_subjects_and_title(doc_title)` _index_earliest=-61m@m _index_latest=-1m@m \
+| rex field=owner "[^@]+@(?<src_domain>[^@]+)" \
+| rex field=email "[^@]+@(?<dest_domain>[^@]+)" \
+| where NOT src_domain=`cs_email_user_domain` and dest_domain=`cs_email_user_domain` \
+| stats values(doc_title) as doc_title, values(doc_type) as doc_type, count, min(_time) as firstTime, max(_time) as lastTime by owner, email \
+| rename email as external_email \
+| eval cyences_severity = "low" \
+| `cs_human_readable_time_format(firstTime)` \
+| `cs_human_readable_time_format(lastTime)` \
+| `cs_gws_suspicious_file_shared_by_external_user_filter`
+action.cyences_notable_event_action = 1
+action.cyences_notable_event_action.param.filter_macro_name = cs_gws_suspicious_file_shared_by_external_user_filter
+action.cyences_notable_event_action.contributing_events = `cs_gsuite` sourcetype="gws:reports:drive" NOT email IN ("", "null") `cs_email_suspicious_subjects_and_title(doc_title)` | rex field=owner "[^@]+@(?<src_domain>[^@]+)" | rex field=email "[^@]+@(?<dest_domain>[^@]+)" | where NOT src_domain=`cs_email_user_domain` and dest_domain=`cs_email_user_domain` | rename email as external_email 
+action.cyences_notable_event_action.system_compromised_search = | stats count by owner, external_email
+action.cyences_notable_event_action.system_compromised_drilldown = `cs_gsuite` sourcetype="gws:reports:drive" NOT email IN ("", "null") `cs_email_suspicious_subjects_and_title(doc_title)` | rex field=owner "[^@]+@(?<src_domain>[^@]+)" | rex field=email "[^@]+@(?<dest_domain>[^@]+)" | where NOT src_domain=`cs_email_user_domain` and dest_domain=`cs_email_user_domain` | search owner=$row.owner$ email=$row.external_email$
```

Comment:
Add src_domain and dest_domain extraction in props.conf to avoid writing such big query here.

Reply:
src_domain and dest_domain is extracted from owner and email fields respectively. & owner and email fields are extracted using EVAL statement so to extract the src_domain and dest_domain fields using rex require higher precedence then EVAL.

hence need to add it as part of search only.